client/src/research/InteractionsView.js

```diff
@@ -45,27 +76,51 @@ class InteractionsView extends Component {
     return sum/interactions.length*100;
   }
 
-  renderPercentRightPerProfile(interactions){
-    var totals = {};
-    var percents = {};
+  renderPercentRightPerProfile(interactions, key){
+    const percents = this.percentRightPerProfile(interactions, key);
+    return <pre> {JSON.stringify(percents, null,2)} </pre>;
+  }
+  renderBarChart(interactions, key, title){
+    const p = this.percentRightPerProfile(interactions, key);
+    const swipes = this.totalSwipes(interactions, key); 
+    var keys=[]; 
+    var d =[]; 
+    var count = 0;
+    var values = [];
```

Comment:
The variable names can help clarify what the purpose of this bit of code is - I think it's ultimately making two things: a list of labels and a list of data points for the chart.  Is there a way to revise this so that the code reads like that?